FxAClient/Frontend/SignIn/FxASignInWebViewController.swift

```diff
@@ -0,0 +1,179 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import UIKit
+import WebKit
+
+protocol FxASignInWebViewControllerDelegate {
+    func didCancel() -> Void
+    func didLoad() -> Void
+    func didSignIn(data: JSON) -> Void
+}
+
+/**
+ * A controller that manages a single web view connected to the Firefox
+ * Accounts (Desktop) Sync postMessage interface.
+ *
+ * The postMessage interface is not really documented, but it is simple
+ * enough.  I reverse engineered it from the Desktop Firefox code and the
+ * fxa-content-server git repository.
+ */
+class FxASignInWebViewController: UIViewController, WKScriptMessageHandler {
+    var delegate: FxASignInWebViewControllerDelegate?
+
+    // Optional delegate for detecting failed requests.
+    var navigationDelegate: WKNavigationDelegate?
+
+    var url: NSURL?
+
+    private var webView: WKWebView!
+
+    override func loadView() {
+        NSLog("loadView!")
+        super.loadView()
+    }
+
+    func startLoad(url: NSURL) {
+        NSLog("startLoad")
+        self.url = url
+        loadWebView(url)
+        NSLog("startLoad: \(webView)")
+    }
+
+    override func viewDidLoad() {
+        NSLog("viewDidLoad!")
+        super.viewDidLoad()
+
+        navigationItem.leftBarButtonItem = UIBarButtonItem(barButtonSystemItem: UIBarButtonSystemItem.Cancel, target: self, action: "didCancel")
+
+        view.addSubview(webView!)
+
+        // Web content fills view.
+        webView.snp_makeConstraints { make in
+            make.edges.equalTo(self.view)
+            return
+        }
+    }
+
+    override func viewDidDisappear(animated: Bool) {
+        super.viewDidDisappear(animated)
+        webView = nil
```

Comment:
I don't think you need to to do this because when your viewcontroller is popped of the navigationvc, it will be deallocated for you. It also makes not difference since you added the webView as a subview, so you are just nilling out webView but it will still be alive in the view hierarchy.

I think past iOS6 it is very uncommon to manually destroy views part of a viewcontroller.